test/e2e/lib/jest/globalSetup.ts

```diff
@@ -33,6 +39,32 @@ export default async (): Promise< void > => {
 	const userList = [ 'gutenbergSimpleSiteUser', 'eCommerceUser', 'defaultUser' ];
 
 	for await ( const user of userList ) {
+		const cookiePath = path.join( cookieBasePath, `${ user }.json` );
+
+		/*
+		This section is nearly identical to the snippet in `BrowserManager.setLoginCookie`, but
+		instead of checking that cookies are older than 3 days, this checks whether the cookies
+		are newer than 3 days. If so, presumably the cookies are still good to use and the iteration
+		is short-circuited.
+		*/
+		try {
+			await access( cookiePath );
+			const stats = await stat( cookiePath );
+			const createdTime = Math.trunc( stats.birthtimeMs );
+			const currentTime = new Date();
+
+			if ( createdTime > currentTime.setDate( currentTime.getDate() - 3 ) ) {
+				console.log( `\nCookie file on disk for ${ user } is recent, skipping.` );
+				continue;
+			} else {
+				// Remove the stale cookie file as BrowserContext.storageState does not
+				// overwrite files.
+				await unlink( cookiePath );
```

Comment:
Either `BrowserContext.storageState` doesn't overwrite the file, or the overwritten file still has the `stat.birthTimeMS` of the old version of the file therefore is regarded as a 'stale' cookie.